test/e2e/kuma_deploy_multi_apps_test.go

```diff
@@ -0,0 +1,75 @@
+package e2e_test
+
+import (
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Kong/kuma/test/framework"
+)
+
+var _ = Describe("Test App deployment", func() {
+
+	var clusters framework.Clusters
+
+	BeforeEach(func() {
+		var err error
+		clusters, err = framework.NewK8sClusters(
+			[]string{framework.Kuma1},
+			framework.Verbose)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = clusters.CreateNamespace("kuma-test")
+		Expect(err).ToNot(HaveOccurred())
+
+		err = clusters.LabelNamespaceForSidecarInjection("kuma-test")
+		Expect(err).ToNot(HaveOccurred())
+
+		err = clusters.DeployKuma()
+		Expect(err).ToNot(HaveOccurred())
+
+		err = clusters.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		err := clusters.DeleteNamespace("kuma-test")
+		Expect(err).ToNot(HaveOccurred())
+
+		_ = clusters.DeleteKuma()
+	})
+
+	It("Should deploy two apps", func() {
+		// given
+		c := clusters.GetCluster(framework.Kuma1)
+
+		// when
+		err := c.DeployApp("kuma-test", "example-app")
+		Expect(err).ToNot(HaveOccurred())
+
+		err = c.DeployApp("kuma-test", "example-client")
+		Expect(err).ToNot(HaveOccurred())
+
+		clientPods := k8s.ListPods(c.GetTesting(),
+			c.GetKubectlOptions("kuma-test"),
+			metav1.ListOptions{
+				LabelSelector: "app=example-client",
+			})
+		Expect(len(clientPods)).To(Equal(1))
+
+		clientPod := clientPods[0]
+
+		k8s.WaitUntilPodAvailable(c.GetTesting(),
+			c.GetKubectlOptions("kuma-test"),
+			clientPod.GetName(),
+			defaultRetries, defaultTimeout)
```

Comment:
Should `WaitUntilPodAvailable` be in `DeployApp`?

Reply:
Yes, we can do that but in a follow-up PR and when needed. As discussed, this can also be a more complicated method to return and interface which can provide various operations on the deployed app (like execute commands withing the pods)